Xamarin.Forms.Core/TypedBinding.cs

```diff
@@ -272,7 +272,19 @@ void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
 			{
 				if (!string.IsNullOrEmpty(e.PropertyName) && string.CompareOrdinal(e.PropertyName, PropertyName) != 0)
 					return;
-				Device.BeginInvokeOnMainThread(() => _binding.Apply(false));
+
+				if (sender is BindableObject element && element.Dispatcher.IsInvokeRequired)
```

Comment:
oh, you even did that ? 👍

Reply:
I feel that this 3way `if` is a kind of a trend, and could be extracted in your DispatcherExtensions ?

Reply:
Done - it's all in one place now.